### DISCUSSION TITLE
How can I test the connection refused case?

### DISCUSSION THREAD

**Discussion by LonelyGriffin** [pactumjs/pactum#62] [Q&A] [answered]
I would like to write something like that:
```js
await server.run()
await pactum.spec().get(healthUrl).expectStatus(200).expectBody('OK')

await server.shutdown()
await pactum.spec().get(healthUrl).expectConnectionRefused()
```

But I have not found a similar  to **expectConnectionRefused**  method. So I try to write something like that:
```js
await server.run()
await pactum.spec().get(healthUrl).expectStatus(200).expectBody('OK')

await server.shutdown()
await pactum
  .spec()
  .get(healthUrl)
  .expect(({res}) => {
    expect(res.code).toBe('ECONNREFUSED')
  })
```

The problem with this way is the logger.  I get the error log. Like this:
![image](https://user-images.githubusercontent.com/7679927/116395224-864aab80-a82c-11eb-8b68-11c6de0e3a69.png)

I can suppress this error, but I think it is better to have a native method for this case, or is there already a better solution?

**Comment by ASaiAnudeep** [comment, marked as answer]
###### Welcome to PactumJS Discussions & Thanks a lot for using this library.

This is an interesting use case. PactumJS doesn't have a native method to support error expectations. I too agree that we should have one.

As you mentioned having a method like `expectConnectionRefused` would solve this issue but it is explicit. We will discuss further and consider adding this method.

Instead, I will add a method something like `expectError` that would be generic for broader use cases.

```js
// expect any error
await pactum.spec().get(healthUrl).expectError();

// expect ECONNREFUSED error
await pactum.spec().get(healthUrl).expectError('ECONNREFUSED');

// expect error object
await pactum.spec().get(healthUrl).expectError({ code: 'ECONNREFUSED' });
```

And once again thank you for your valuable suggestion.

**Reply by ASaiAnudeep** [reply]
@LonelyGriffin - `expectError` method is published as part of **3.0.12** version.
